crates/astria-core/src/sequencer/v1alpha1/block.rs

```diff
@@ -366,6 +393,25 @@ impl SequencerBlock {
         }
     }
 
+    #[must_use]
+    pub fn filtered_block(self, rollup_ids: Vec<RollupId>) -> Self {
```

Comment:
The naming of this method does not feel idiomatic:
1. manipulations of the same type should have a verb in the name, e.g. `filter_blocks_by_rollup_id` or something like that.
2. transformations to a different type should be `to_*` or `into_*`
3. This returning `Self` feels wrong. I'd prefer to call it `FilteredSequencerBlock` or `SequencerBlockSubset` or something like that
4. I'd expect that a function named `filter` takes a function `F: FnMut(&RollupId) -> bool` to match the ecosystem. But even if it doesn't a better signature would be `f<T: IntoIterator<Item = RollupId>>(self, rollup_ids: T)` or at least `f(self, rollup_ids: &[RollupId])`

Reply:
updated to be `into_filtered_block`, thoughts?